components/infobox/wikis/rainbowsix/infobox_league_custom.lua

```diff
@@ -0,0 +1,273 @@
+			table.insert(widgets,
+				Cell{
+					name = 'Ubisoft Tier',
+					content = {'[['.._UBISOFT_TIERS[args.ubisofttier:lower()]..']]'},
+					classes = {'valvepremier-highlighted'}
+				}
+			)
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:addToLpdb(lpdbData, args)
+	lpdbData.maps = table.concat(_league:getAllArgsForBase(args, 'map'), ';')
+
+	lpdbData.publishertier = args.ubisofttier:lower()
+	lpdbData.participantsnumber = args.player_number or args.team_number
+	lpdbData.extradata = {
+		individual = String.isNotEmpty(args.player_number) and 'true' or '',
+		startdatetext = CustomLeague:_standardiseRawDate(args.sdate or args.date),
+		enddatetext = CustomLeague:_standardiseRawDate(args.edate or args.date),
+	}
+
+	return lpdbData
+end
+
+function CustomLeague:_standardiseRawDate(dateString)
+	-- Length 7 = YYYY-MM
+	-- Length 10 = YYYY-MM-??
+	if String.isEmpty(dateString) or (#dateString ~= 7 and #dateString ~= 10) then
+		return ''
+	end
+
+	if #dateString == 7 then
+		dateString = dateString .. '-??'
+	end
+	dateString = dateString:gsub('-XX', '-??')
+	return dateString
+end
+
+function CustomLeague:_createPrizepool()
+	if String.isEmpty(_args.prizepool) and String.isEmpty(_args.prizepoolusd) then
+		return nil
+	end
+	local date
+	if String.isNotEmpty(_args.currency_rate) then
+		date = _args.currency_date
+	end
+
+	return PrizePoolCurrency._get({
+		prizepool = _args.prizepool,
+		prizepoolusd = _args.prizepoolusd,
+		currency = _args.localcurrency,
+		rate = _args.currency_rate,
+		date = date or Variables.varDefault('tournament_enddate', _TODAY),
+	})
+end
+
+function CustomLeague:_createLiquipediaTierDisplay()
+	local tier = _args.liquipediatier or ''
+	local tierType = _args.liquipediatiertype or ''
+	if String.isEmpty(tier) then
+		return nil
+	end
+
+	local function buildTierString(tierString)
+		local tierText = Tier.text[tierString]
+		if not tierText then
+			table.insert(_league.warnings, tierString .. ' is not a known Liquipedia Tier/Tiertype')
+			return ''
+		else
+			return '[[' .. tierText .. ' Tournaments|' .. tierText .. ']][[Category:' .. tierText .. ' Tournaments]]'
+		end
+	end
+
+	local tierDisplay = buildTierString(tier)
+
+	if String.isNotEmpty(tierType) then
+		tierDisplay = buildTierString(tierType) .. '&nbsp;(' .. tierDisplay .. ')'
+	end
+
+	return tierDisplay
+end
+
+function CustomLeague:defineCustomPageVariables()
+	--Legacy vars
+	Variables.varDefine('tournament_ticker_name', _args.tickername or '')
+	Variables.varDefine('tournament_tier', _args.liquipediatier or '')
+	Variables.varDefine('tournament_tier_type', _args.liquipediatiertype or '')
+	Variables.varDefine('tournament_ticker_name', _args.tickername or '')
+	Variables.varDefine('tournament_prizepool', _args.prizepool or '')
+	Variables.varDefine('tournament_mode', _args.mode or '')
+	Variables.varDefine('tournament_currency', _args.currency or '')
```

Comment:
isn't that already set by the prizepoolcurrency module?

Reply:
Prize pool currency will usually set this variable.
However the module won't be run if certain arguments are not yet known (prizepool or prizepoolusd)

Reply:
okay